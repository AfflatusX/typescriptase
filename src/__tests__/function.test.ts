import { function1, function2, function3 } from "./setup";

test(
  "function1",
  async (): Promise<void> => {
    expect(function1.render({
      name: "function1",
      path: "src/__tests__/function.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "function2",
  async (): Promise<void> => {
    expect(function2.render({
      name: "function2",
      path: "src/__tests__/function.test.ts",
    }))
      .toMatchSnapshot();
  },
);

test(
  "function3",
  async (): Promise<void> => {
    expect(function3.render({
      name: "function3",
      path: "src/__tests__/function.test.ts",
    }))
      .toMatchSnapshot();
  },
);
