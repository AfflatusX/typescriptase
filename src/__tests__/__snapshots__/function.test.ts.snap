// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function1 1`] = `
"
function function1(
): void {
}
"
`;

exports[`function2 1`] = `
"
export async function function2(
): Promise<string> {
}
"
`;

exports[`function3 1`] = `
"
export async function function2(
  var1: string,
  var2: object,
): Promise<string[]> {
  /* BESPOKE START <<fn1>> */
  /* BESPOKE END <<fn1>> */
}
"
`;
